src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,415 @@
+        public virtual LocalizedString this[[NotNull] string key] => _localizer[key];
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key, params object[] arguments] =>
+            _localizer[key, arguments];
+
+        /// <summary>
+        /// Creates a new <see cref="IStringLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IStringLocalizer"/>.</returns>
+        IStringLocalizer IStringLocalizer.WithCulture([NotNull] CultureInfo culture) => WithCulture(culture);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key) => _localizer.GetString(key);
+
+        /// <inheritdoc />
+        public virtual LocalizedString GetString([NotNull] string key, params object[] arguments) =>
+            _localizer.GetString(key, arguments);
+
+        /// <inheritdoc />
+        public IEnumerable<LocalizedString> GetAllStrings(bool includeAncestorCultures) =>
+            _localizer.GetAllStrings(includeAncestorCultures);
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key) => ToHtmlString(_localizer.GetString(key));
+
+        /// <inheritdoc />
+        public virtual LocalizedHtmlString Html([NotNull] string key, params object[] arguments)
+        {
+            var stringValue = _localizer[key].Value;
+
+            return ToHtmlString(new LocalizedString(key, EncodeArguments(stringValue, arguments)));
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="IHtmlLocalizer"/> for a specific <see cref="CultureInfo"/>.
+        /// </summary>
+        /// <param name="culture">The <see cref="CultureInfo"/> to use.</param>
+        /// <returns>A culture-specific <see cref="IHtmlLocalizer"/>.</returns>
+        public IHtmlLocalizer WithCulture([NotNull] CultureInfo culture) =>
+            new HtmlLocalizer(_localizer.WithCulture(culture), _encoder);
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected LocalizedHtmlString ToHtmlString(LocalizedString result) =>
+            new LocalizedHtmlString(result.Name, result.Value, result.ResourceNotFound);
+
+        /// <summary>
+        /// Creates a <see cref="IStringLocalizer"/>.
+        /// </summary>
+        /// <param name="baseName">The base name of the resource to load strings from.</param>
+        /// <param name="location">The location to load resources from.</param>
+        /// <returns>The <see cref="IStringLocalizer"/>.</returns>
+        protected void CreateStringLocalizer(string baseName, string location) =>
+            _localizer = _localizerFactory.Create(baseName, location);
```

Comment:
Will hit an NRE if `_localizerFactory == null`. Other properties and methods have similar issues if `_localizer == null`. Add the following and use them everywhere to `throw` a better `Exception`:
``` c#
private IStringLocalizer Localizer
{
    get
    {
        if (_localizer == null)
        {
            // Message should be a resource similar to HtmlHelper_NotContextualized.
            throw new InvalidOperationException("Must call CreateStringLocalizer before using this property.");
        }

        return _localizer;
    }
    [param: NotNull]
    set
    {
        _localizer = value;
    }
}

private IStringLocalizerFactory LocalizerFactory
{
    get
    {
        if (_localizerFactory == null)
        {
            // Message should be a resource similar to HtmlHelper_NotContextualized.
            throw new InvalidOperationException("Call other constructor overload to use this property.");
        }

        return _localizerFactory;
    }
    [param: NotNull]
    set
    {
        _localizerFactory = value;
    }
}
```